test/serialization.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-unused-vars */
+'use strict';
+
+const { runInNewContext } = require('vm');
+
+const metatests = require('metatests');
+const metaschema = require('metaschema');
+
+const { serializeSchema } = require('../lib/serialization');
+const { options, config } = require('../lib/metaschema-config/config');
+
+const decorators = require('../lib/metaschema-config/decorators');
+
+metatests.test('Serialization', async test => {
+  const schema = await metaschema.fs.load(
+    'test/fixtures/serialization',
+    options,
+    config
+  );
+
+  const domains = schema.schemas[0];
+  const stringifyDomains = serializeSchema(domains, {
+    exclude: ['definition'],
+  });
+
+  const category = schema.categories.get('Test');
+  const { type, module, name, source } = serializeSchema(category, {
+    exclude: ['definition'],
+  });
+  const stringifyCategory = { type, name, module, source };
+
+  const action = schema.actions.get('Test');
+  const stringifyAction = serializeSchema(action, {
+    exclude: ['definition'],
+  });
+
+  stringifyDomains.definition = metaschema.processSchema(
+    'TestDomains',
+    stringifyDomains.source,
+    {
+      decorators: decorators.localDecorators.domains,
+    },
+    runInNewContext
+  );
+
+  stringifyCategory.definition = metaschema.processSchema(
+    'TestCategory',
+    stringifyCategory.source,
+    {
+      decorators: decorators.localDecorators.category,
+    },
+    runInNewContext
+  );
+
+  stringifyAction.definition = metaschema.processSchema(
+    'TestAction',
+    stringifyAction.source,
+    {
+      decorators: {
+        ...decorators.localDecorators.action,
+        ...decorators.decorators,
+      },
+    },
+    runInNewContext
+  );
+
+  const catalog = schema.categories.get('Catalog');
+  const subsystem = schema.categories.get('Subsystem');
+
+  const newSchema = metaschema.Metaschema.create(
+    [stringifyDomains, stringifyCategory, stringifyAction, catalog, subsystem],
+    config
+  );
+
+  const newDomains = newSchema.schemas[0];
```

Comment:
You should not rely on load order here:
```suggestion
  const newDomains = newSchema.schemas.find(schema => schema.type === 'domains');
```